nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/QueryDatabaseTable.java

```diff
@@ -179,6 +189,9 @@ public void onTrigger(final ProcessContext context, final ProcessSessionFactory
         final String maxValueColumnNames = context.getProperty(MAX_VALUE_COLUMN_NAMES).getValue();
         final Integer fetchSize = context.getProperty(FETCH_SIZE).asInteger();
         final Integer maxRowsPerFlowFile = context.getProperty(MAX_ROWS_PER_FLOW_FILE).asInteger();
+        final Integer maxFragments = context.getProperty(MAX_FRAGMENTS).isSet()
+                ? context.getProperty(MAX_FRAGMENTS).asInteger()
+                : 0;
```

Comment:
Is zero a valid number of fragments on its own (versus leaving the property blank)? If so then perhaps the "not set" value should be -1 and the logic below changed to >= 0. Otherwise perhaps the default value for the property should be zero and the description updated to reflect that zero means all fragments.